src/mlir/README.md

```diff
@@ -0,0 +1,23 @@
+# Verona MLIR Dialect
+
+This is a prototype for the Verona MLIR dialect.
+
+## Building LLVM
+
+Build LLVM with the following options:
+
+```bash
+$ cmake -G Ninja ../llvm \
+        -DLLVM_ENABLE_PROJECTS=mlir \
+        -DLLVM_TARGETS_TO_BUILD="X86" \
+        -DCMAKE_BUILD_TYPE=Debug \
+        -DLLVM_ENABLE_ASSERTIONS=ON \
+        -DCMAKE_C_COMPILER=clang \
+        -DCMAKE_CXX_COMPILER=clang++ \
+        -DLLVM_ENABLE_LLD=ON \
+        -DLLVM_REQUIRES_EH=ON \
```

Comment:
This is required by us. I may be building it wrong, but some external dependencies have exception handling and don't link well against LLVM's. I haven't dug deeply, tbh. Would be very happy if we don't need to do that.

Reply:
Just to note that, currently, the MLIR generator uses EH, but that's a stop gap until we have decent error handling.

Reply:
In the past, I used non-RTTI builds of LLVM with a front end that used RTTI and exceptions.  Everything was fine unless you subclassed an LLVM class.  I wonder if we have the same issue, not passing -fno-rtti to the builds of our MLIR things?

Someone accidentally missed off some anchors in LLVM's Instructions.h, which caused the vtable for a couple of LLVM classes to be emitted in every compilation unit, which was a problem, but I think that was fixed a few years back.

Reply:
We do subclass a lot of MLIR stuff for the dialect. I really want to get rid of EH+RTTI, by probably using `llvm::ErrorOr`, but I haven't got around using it. The main reason is that I don't just want to treat errors as they happen, I want to compose a stack trace of errors and let the top caller handle it. This will be fundamental for partial compilation, pretty printing, IDE integration, etc.